card/mod/standard/set/all/links.rb

```diff
@@ -1,81 +1,93 @@
+    content_tag :a, text, opts
+  end
+
+  def interpret_data_opts_to_link_to opts
+    [:remote, :method].each do |key|
+      next unless (val = opts.delete key)
+      opts["data-#{key}"] = val
+    end
+  end
+end
+
+format :css do
+  def link_to _text=nil, opts={}
+    card_url interpret_pathish(opts.delete(:path))
+  end
+end
+
+format do
+  def link_to text=nil, opts={}
+    href = interpret_pathish opts.delete(:path)
     if text && href != text
       "#{text}[#{href}]"
     else
       href
     end
   end
 
-  # link to url, view, card or related card
-  def smart_link link_text, target, html_args={}
-    if (view = target.delete(:view))
-      view_link link_text, view, html_args.merge(path_opts: target)
-    elsif (page = target.delete(:card))
-      card_link page, html_args.merge(path_opts: target, text: link_text)
-    elsif target[:related]
-      if target[:related].is_a? String
-        target[:related] = { name: "+#{target[:related]}" }
-      end
-      view_link link_text, :related, html_args.merge(path_opts: target)
-    elsif target[:web]
+  def smart_link_to text, opts={}
+    if (linktype = [:view, :related, :card, :resource].find { |key| opts[key] })
+      send "link_to_#{linktype}", opts.delete(linktype), text, opts
     else
-      link_to link_text, target, html_args
+      send :link_to, text, opts
     end
   end
 
-  # link to a specific url or path
-  def web_link href, opts={}
-    text = opts.delete(:text) || href
-    new_class =
-      case href
-      when /^https?\:/
-        opts[:target] = "_blank"
-        "external-link"
-      when %r{^/}
-        href = internal_url href[1..-1]
-        "internal-link"
-      when /^mailto\:/                    then "email-link"
-      when /^([a-zA-Z][\-+\.a-zA-Z\d]*):/ then Regexp.last_match(1) + "-link"
-      else                                return card_link href, opts
-      end
-    add_class opts, new_class
-    link_to text, href, opts
-  end
-
-  # link to a specific card
-  def card_link name_or_card, opts={}
-    name = Card::Name.cardish name_or_card
-    text = (opts.delete(:text) || name).to_name.to_show @context_names
-
-    path_opts = opts.delete(:path_opts) || {}
-    path_opts[:name] = name
-    known = opts[:known].nil? ? Card.known?(name) : opts.delete(:known)
+  def link_to_resource resource, text=nil, opts={}
+    case (resource_type = resource_type resource)
+    when "external-link" then opts[:target] = "_blank"
+    when "internal-link" then resource = internal_url resource[1..-1]
+    end
+    add_class opts, resource_type
+    link_to text, opts.merge(path: resource)
+  end
+
+  def resource_type resource
+    case resource
+    when /^https?\:/          then "external-link"
+    when %r{^/}               then "internal-link"
+    when /^mailto\:/          then "email-link"
+    when RESOURCE_TYPE_REGEXP then Regexp.last_match(1) + "-link"
+    end
+  end
+
+  def link_to_card cardish, text=nil, opts={}
```

Comment:
`link_to_card cardish, text_or_opts={}, opts=nil` to avoid the nil for card links with the name as text?